src/readahead/readahead-collect.c

```diff
@@ -253,9 +253,17 @@ static int collect(const char *root) {
 
         assert(root);
 
-        if (asprintf(&pack_fn, "%s/.readahead", root) < 0) {
+        if (arg_pack_file) {
+            pack_fn = strdup(arg_pack_file);
+            if (!pack_fn) {
                 r = log_oom();
                 goto finish;
+            }
+        } else {
```

Comment:
Wonder if you couldn't combine these branches by setting `pack_fn` when `arg_pack_file` is `NULL`, so not to have two similar checks (also below)